homeassistant/components/sonos/__init__.py

```diff
@@ -84,6 +84,8 @@ def __init__(self) -> None:
         self.favorites: dict[str, SonosFavorites] = {}
         self.alarms: dict[str, SonosAlarms] = {}
         self.topology_condition = asyncio.Condition()
+        self.unjoin_events: dict[str, asyncio.Event] = {}
+        self.unjoin_speakers: dict[str, list[SonosSpeaker]] = {}
```

Comment:
Theoretically these 2 dicts could end up being out of sync. I would suggest to use a namedtuple / data class that holds an event and a speaker.